filesystem/src/main/java/com/nytimes/android/external/fs/filesystem/FSFile.java

```
@@ -71,6 +71,9 @@ public void write(BufferedSource source) throws IOException {
             if (sink != null) {
                 sink.close();
             }
+            if (source != null) {
+                source.close();
+            }
```

in future we may want to use the ` try-with-resources ` syntax to handle the `.close()`  and avoid need for the `finally`  block.
From what I read this only works on minSdk 19+ (not sure if that is a concern)

In this case we still want to delete the `tmpFile` so still need the `finally`

---

The minSdk bit would probably be cause for concern.

---

Why the null check? Makes more sense to me to annotate the source param as `@NonNull`. The whole method makes no sense with a null source and it should throw an exception if it's called with wrong arguments.

---

@PaulWoitaschek That's a fair point. Annotated the arg and removed the null check.